src/definitions/behaviors/api.js

```diff
@@ -868,7 +868,7 @@ $.api = $.fn.api = function(parameters) {
           },
           templatedURL: function(action) {
             action = action || $module.data(metadata.action) || settings.action || false;
-            url    = $module.data(metadata.url) || settings.url || false;
+            url    = settings.url || $module.data(metadata.url) || false;
```

Comment:
what about the config one line above, shouldn't it be fixed too (and maybe there are also more places)

Reply:
👍🏼 You've got a point here (i was so concentrated on your issue with the url 😆 
Will check the reamining code and ajust accordingly. Thanks for the hint

Reply:
Done by [1b75224](https://github.com/fomantic/Fomantic-UI/pull/2582/commits/1b752241e4d26c5123da3fcf711d1fa7a09edc91) Seems to be the only other similiar code in the whole api

Reply:
`(?<!meta)data.*setti` regex returns 27 matches

- I am not sure about https://github.com/fomantic/Fomantic-UI/blob/7071e5977705394994b87df42855435f4b357920/src/definitions/behaviors/api.js#L205
- in calendar and popup module there is quite simillar pattern like in API

Reply:
> 

In API only? When just searching for module.data and also same settings attribute (in API.js) i cannot find any similar statement 

> - I am not sure about https://github.com/fomantic/Fomantic-UI/blob/7071e5977705394994b87df42855435f4b357920/src/definitions/behaviors/api.js#L205

This is correct, as it was implemented some versions ago to respect already sent data

> - in calendar and popup module there is quite simillar pattern like in API

Well, your issue was initially about API..;)... i take a look at the other modules as well

Reply:
> `(?<!meta)data.*setti` regex returns 27 matches
> * in calendar and popup module there is quite simillar pattern like in API

i checked the source given your search regex. I could not identify the same issues in the other modules because the found targets were either 
- get functions where is is intended to get a possibly existing previously created metadata before
- currently not getting any data at all for metadate as the necessary code for setting the date is missing (so it always fetches the setting)
- having  a non falsy default setting (like a string) which would mean we _always_ would provide the setting and never catch an existing .data setting
- did not apply at all 

That said, i am merging this PR as is. If you think we are still have similiar issues in other modules, please prepare a jsfiddle testcase 🙂

Reply:
> If you think we are still have similiar issues in other modules, please prepare a jsfiddle testcase 🙂

No, I am not aware of the same problem anywhere else, I am just used to check every similar pattern (often using regex) everywhere else.